crates/lsp/src/handlers_state.rs

```diff
@@ -62,7 +63,13 @@ pub(crate) fn initialize(
     params: InitializeParams,
     lsp_state: &mut LspState,
     state: &mut WorldState,
+    log_tx: LogMessageSender,
 ) -> anyhow::Result<InitializeResult> {
+    // TODO: Get user specified log level from `params.initialization_options`
+    let log_level = None;
+
+    logging::init_logging(log_tx, log_level, params.client_info.as_ref());
```

Comment:
Initializing logging in the `initialize()` handler is as early as I think we can initialize it. We are going to require information from the Client to correctly set up our logging infrastructure, so we can't do this any earlier.